cpu/cc2538/dev/cc2538-aes-128.c

```diff
@@ -82,6 +83,7 @@ set_key(const uint8_t *key)
                       CC2538_AES_128_KEY_AREA);
   if(ret != CRYPTO_SUCCESS) {
     PRINTF("%s: aes_load_keys() error %u\n", MODULE_NAME, ret);
+    watchdog_reboot();
```

Comment:
Rethinking about this, it's actually better to use `sys_ctrl_reset()` from `cpu/cc2538/dev/sys-ctrl.h` for CC2538-specific code because:
 - It's immediate (vs. Twdt x 64).
 - `SYS_CTRL_CLOCK_STA.RST` then indicates `CLD/SW` instead of `WDT`, so it's possible to know that the reset was intentionally triggered by software, rather than by the watchdog timer because of the software accidentally getting out of control.

Out of curiosity, I've looked at what `rom_util_reset_device()` from `cpu/cc2538/dev/rom-util.h` does by disassembling the ROM, and it actually uses an undocumented hardware feature (reserved bit):
```
SYS_CTRL_CLOCK_CTRL = 0x8;
while (1)
  ;
```
According to http://www.ti.com/lit/pdf/swru333, this `performs a system reset equivalent to an external pin reset`, so `sys_ctrl_reset()` is still more informative here.

Note that there is yet another means of resetting the CC2538: `NVIC_APINT.SYSRESETREQ = 1`.